src/move_basic.cpp

```diff
@@ -754,11 +754,11 @@ bool MoveBasic::moveLinear(tf2::Transform& goalInDriving,
         /* Since we are dealing with imperfect localization we should make
          * sure we are at least runawayTimeout driving away from the goal*/
         double angleRemaining = std::atan2(remaining.y(), remaining.x());
-        if (std::cos(angleRemaining) < 0) {
+        if (std::cos(angleRemaining) < 0) { // 0
             if (ros::Time::now() - last > runawayTimeout) {
-                abortGoal("MoveBasic: Moving away from goal");
-                sendCmd(0, 0);
-                return false;
+                ROS_INFO("MoveBasic: Moving away from goal");
+                sendCmd(rotation, 0);
```

Comment:
Rotation in linear part(moveLinear function), does not relate to rotation when doing rotation part(rotate function). You should do: sendCmd(0,0);
When moveLinear function returns true, the rotate function is called again to turn the robot in the orientation that is specified in the goal send to move_basic.

Reply:
Ah, so that's the rotation that happens while you're driving ahead and it's doing small corrections? Ok I'll set it to 0,0 and re-test, but it should behave the same way in theory?

Reply:
Yes, the moveLinear function will return and then the rotate function will send new velocity commands